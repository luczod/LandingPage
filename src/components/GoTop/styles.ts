import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    position: fixed;
    right: 3rem;
    bottom: 2rem;
    z-index: 6;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 4rem;
    width: 4rem;
    opacity: 0.8;
  `}
`;
