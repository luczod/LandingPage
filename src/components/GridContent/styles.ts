import styled, { DefaultTheme, css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: 0;
    margin: 0 auto;
  `}
`;

export const HtmlWrap = styled.div`
  ${({ theme }) => css`
    margin: 0;
  `}
`;
