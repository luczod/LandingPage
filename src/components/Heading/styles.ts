import styled, { DefaultTheme, css } from "styled-components";
type OnlyKeys = keyof typeof titleSize;
interface ISProps {
  $colorark: boolean;
  size: OnlyKeys;
  $uppertitle?: boolean;
}

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.md};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.lg};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xl};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhg};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xl};
  }
`;

export const Title = styled.h1<ISProps>`
  ${({ theme, size, $uppertitle, $colorark: colorDark }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    text-transform: ${$uppertitle ? "uppercase" : "none"};
  `}
`;
