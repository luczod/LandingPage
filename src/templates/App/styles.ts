import styled, { css } from "styled-components";

interface IProps {
  background?: string;
}

export const Wrapper = styled.div<IProps>`
  ${({ background }) => css`
    background: ${background};
  `}
`;
