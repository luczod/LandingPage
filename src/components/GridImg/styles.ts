import styled, { css } from "styled-components";
import { Container as TextComponent } from "../Textfn/styles";

interface IPros {
  open?: boolean;
}

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
  `}
`;

export const GridItem = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 1rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;

export const ModalImg = styled.div`
  ${({ theme }) => css`
    display: none; /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    padding-top: 100px; /* Location of the box */
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0 0 0 / 0.3); /* color-Black w/ opacity */
    border-radius: 1rem;

    & > span {
      position: fixed;
      cursor: pointer;
      top: 4rem;
      right: 2rem;
      color: ${theme.colors.secondaryColor};
      font-size: ${theme.font.sizes.hg};
      &::hover {
        color: ${theme.colors.white};
      }
      @media only screen and (max-width: 768px) {
        left: 1rem;
      }
    }

    & > img {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      border-radius: 1rem;
      /* 100% Image Width on Smaller Screens */
      @media only screen and (max-width: 768px) {
        width: 100%;
        padding: 0 0.8rem;
      }
    }
    & > div {
      margin: auto;
      display: block;
      width: 80%;
      max-width: 700px;
      text-align: center;
      font-size: ${theme.font.sizes.md};
      color: ${theme.colors.white};
      padding: 10px 0;
      height: 150px;
    }
  `}
`;
