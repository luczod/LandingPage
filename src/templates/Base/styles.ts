import styled, { css } from "styled-components";
import { Container as SectionContainer } from "../../components/SectionContainer/styles";
import { Container as Textfn } from "../../components/Textfn/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }

    & ${Textfn} {
      font-size: ${theme.font.sizes.sm};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
