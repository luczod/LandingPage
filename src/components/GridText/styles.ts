import styled, { css } from "styled-components";
import { Title as ItemTitle } from "../Heading/styles";
import { Container as TextComponent } from "../Textfn/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.lg};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    overflow: hidden;
    width: 100%;
    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridItem = styled.article`
  ${({ theme }) => css`
    ${ItemTitle} {
      position: relative;
      left: 5rem;
    }

    ${ItemTitle}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhg};
      top: -2rem;
      left: -5rem;
      transform: rotate(8deg);
    }
  `}
`;
