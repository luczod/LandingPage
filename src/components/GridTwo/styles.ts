import styled, { DefaultTheme, css } from "styled-components";
import { Title } from "../Heading/styles";

interface IProps {
  background: boolean;
}

export const Container = styled.div<IProps>`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextColumn = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.lg};
      text-align: center;
    }
  `}
`;

export const ImgWrap = styled.div`
  ${({ theme }) => css``}
`;

export const Imgfn = styled.img`
  ${({ theme }) => css``}
`;
