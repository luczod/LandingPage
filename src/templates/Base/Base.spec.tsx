import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Base } from ".";

describe("<Base/>", () => {
  it("should render", () => {
    renderTheme(<Base>Texto</Base>);
    expect(screen.getByRole("heading")).toBeInTheDocument();
  });
});
