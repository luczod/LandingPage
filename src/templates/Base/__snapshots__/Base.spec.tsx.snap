// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Base/> > should render base component 1`] = `
.c10 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

.c4 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c13 {
  color: #FFFFFF;
  font-size: 4.0rem;
  text-transform: uppercase;
}

.c18 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c21 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c28 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: uppercase;
}

.c1 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  transition: all 300ms ease-in-out;
  background: #FFFFFF;
  border: 0.1rem solid #DDDDDD;
}

.c1 >.c9 {
  padding-top: 0;
  padding-bottom: 0;
}

.c1 .c3 {
  margin-top: 0;
  margin-bottom: 0;
}

.c2 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-left: 20rem;
  margin-right: 20rem;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 3rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 >svg {
  cursor: pointer;
  padding-top: 2px;
  padding-left: 6px;
  padding-right: 6px;
}

.c5 {
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;
}

.c5 >img {
  height: 6rem;
  width: 14rem;
}

.c6 {
  color: #FFFFFF;
  display: flex;
  flex-flow: row wrap;
}

.c7 {
  color: #0A1128;
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  position: relative;
}

.c7::after {
  content: "";
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c7:hover::after {
  left: 25%;
  width: 50%;
}

.c15 {
  font-size: 2.4rem;
}

.c30 {
  text-align: center;
  border-top: 0.2rem solid #DDDDDD;
}

.c30 a {
  color: inherit;
  text-decoration: none;
}

.c30 .c14 {
  font-size: 1.6rem;
}

.c30 .c9 {
  padding-top: 0;
  padding-bottom: 0;
}

.c31 {
  text-align: center;
}

.c31 a {
  color: inherit;
  text-decoration: none;
  font-size: 1.6rem;
}

.c32 {
  background: #0A1128;
  color: #FFFFFF;
  position: fixed;
  right: 3rem;
  bottom: 2rem;
  z-index: 6;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 4rem;
  width: 4rem;
  opacity: 0.8;
}

.c17 .c14 {
  margin-bottom: 3.2rem;
}

.c19 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
  overflow: hidden;
  width: 100%;
}

.c20 .c3 {
  position: relative;
  left: 5rem;
}

.c20 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  top: -2rem;
  left: -5rem;
  transform: rotate(8deg);
}

.c8 {
  background-color: #FFFFFF;
  color: #0A1128;
  border-bottom: 0.2rem solid #DDDDDD;
  background: #0A1128;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c16 {
  background-color: #FFFFFF;
  color: #0A1128;
  border-bottom: 0.2rem solid #DDDDDD;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c11 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: 3.2rem;
}

.c11 .c3 {
  margin-bottom: 4.0rem;
}

.c27 {
  text-align: center;
  padding: 0;
  margin: 0 auto;
}

.c29 {
  margin: 0;
}

.c22 >.c14 {
  margin-bottom: 3.2rem;
}

.c23 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c24 {
  overflow: hidden;
  border-radius: 1rem;
}

.c25 {
  width: 100%;
  transition: all 300ms ease-in-out;
}

.c25:hover {
  transform: scale(1.2) rotate(10deg);
}

.c26 {
  display: none;
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0 0 0 / 0.3);
  border-radius: 1rem;
}

.c26>span {
  position: fixed;
  cursor: pointer;
  top: 4rem;
  right: 2rem;
  color: #dc143c;
  font-size: 5.6rem;
}

.c26>span::hover {
  color: #FFFFFF;
}

.c26>img {
  margin: auto;
  display: block;
  max-width: 700px;
  object-fit: cover;
  border-radius: 1rem;
}

.c26>div {
  margin: auto;
  display: block;
  width: 80%;
  max-width: 700px;
  text-align: center;
  font-size: 2.4rem;
  color: #FFFFFF;
  padding: 10px 0;
  height: 150px;
}

@media (max-width:768px) {
  .c18 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c1 >.c9 {
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }

  .c1 .c3 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c2 {
    flex-flow: column nowrap;
    justify-content: flex-start;
    align-items: center;
    text-align: center;
    min-height: 100vh;
    height: 100%;
    overflow-y: auto;
    padding: 4.8rem 0;
    cursor: pointer;
  }
}

@media (max-width:768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c6 {
    flex-flow: column wrap;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c19 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c11 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c12 {
    margin-bottom: 3.2rem;
    text-align: center;
  }
}

@media only screen and (max-width: 768px) {
  .c26>span {
    left: 1rem;
  }
}

@media only screen and (max-width: 768px) {
  .c26>img {
    width: 100%;
    padding: 0 0.8rem;
  }
}

<div>
  <button
    aria-label="Open/close menu"
    class="c0"
  >
    <svg
      aria-label="Open menu"
      class="lucide lucide-menu"
      fill="none"
      height="32"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="4"
        x2="20"
        y1="12"
        y2="12"
      />
      <line
        x1="4"
        x2="20"
        y1="6"
        y2="6"
      />
      <line
        x1="4"
        x2="20"
        y1="18"
        y2="18"
      />
    </svg>
  </button>
  <header
    aria-label="wrap"
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3 c4"
      >
        <a
          class="c5"
          href="#"
        >
          Logo
        </a>
      </h1>
      <nav
        aria-label="Main menu"
        class="c6"
      >
        <a
          class="c7"
          href="#target1"
          target="_self"
        >
          url 1
        </a>
        <a
          class="c7"
          href="#target2"
          target="_self"
        >
          url 2
        </a>
        <a
          class="c7"
          href="#target3"
          target="_self"
        >
          url 3
        </a>
        <a
          class="c7"
          href="#target4"
          target="_self"
        >
          url 4
        </a>
        <a
          class="c7"
          href="#target5"
          target="_self"
        >
          url 5
        </a>
        <a
          class="c7"
          href="#target6"
          target="_self"
        >
          url 6
        </a>
        <a
          class="c7"
          href="#target7"
          target="_self"
        >
          url 7
        </a>
        <a
          class="c7"
          href="#target8"
          target="_self"
        >
          url 8
        </a>
        <a
          class="c7"
          href="#target9"
          target="_self"
        >
          url 9
        </a>
        <a
          class="c7"
          href="#target10"
          target="_self"
        >
          url 10
        </a>
      </nav>
    </div>
  </header>
  <section
    class="c8"
  >
    <div
      class="c9 c10"
    >
      <div
        class="c11"
      >
        <article
          class="c12"
        >
          <h1
            class="c3 c13"
          >
            Grid two columns
          </h1>
          <div
            class="c14 c15"
          />
        </article>
        <div
          class=""
        >
          <img
            alt="img"
            class=""
            src="http://localhost:1337assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </section>
  <section
    class="c16"
  >
    <div
      class="c9 c10"
    >
      <div
        class="c17"
      >
        <h1
          class="c3 c18"
        >
          My grid
        </h1>
        <div
          class="c14 c15"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c19"
        >
          <article
            class="c20"
          >
            <h1
              class="c3 c21"
            >
              Teste 1
            </h1>
            <div
              class="c14 c15"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
          <article
            class="c20"
          >
            <h1
              class="c3 c21"
            >
              Teste 2
            </h1>
            <div
              class="c14 c15"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
          <article
            class="c20"
          >
            <h1
              class="c3 c21"
            >
              Teste 3
            </h1>
            <div
              class="c14 c15"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
            </div>
          </article>
        </div>
      </div>
    </div>
  </section>
  <section
    class="c16"
  >
    <div
      class="c9 c10"
    >
      <div
        class="c22"
      >
        <h2
          class="c3 c18"
        >
          My image grid
        </h2>
        <div
          class="c14 c15"
        >
          Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
        </div>
        <div
          class="c23"
        >
          <div
            class="c24"
          >
            <img
              alt="Teste 1"
              aria-label="item-img"
              class="c25"
              src="https://source.unsplash.com/random/800x800?r=1"
            />
            <div
              aria-label="Modal"
              class="c26"
              id="GridImg-8"
            >
              <img
                alt="Teste 1"
                src="https://source.unsplash.com/random/800x800?r=1"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 1
              </div>
            </div>
          </div>
          <div
            class="c24"
          >
            <img
              alt="Teste 2"
              aria-label="item-img"
              class="c25"
              src="https://source.unsplash.com/random/800x800?r=2"
            />
            <div
              aria-label="Modal"
              class="c26"
              id="GridImg-9"
            >
              <img
                alt="Teste 2"
                src="https://source.unsplash.com/random/800x800?r=2"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 2
              </div>
            </div>
          </div>
          <div
            class="c24"
          >
            <img
              alt="Teste 3"
              aria-label="item-img"
              class="c25"
              src="https://source.unsplash.com/random/800x800?r=3"
            />
            <div
              aria-label="Modal"
              class="c26"
              id="GridImg-10"
            >
              <img
                alt="Teste 3"
                src="https://source.unsplash.com/random/800x800?r=3"
              />
              <span
                aria-label="Close-img"
              >
                ×
              </span>
              <div>
                Teste 3
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="c16"
  >
    <div
      class="c9 c10"
    >
      <div
        class="c27"
      >
        <h1
          class="c3 c28"
        >
          Grid content
        </h1>
        <div
          class="c29"
        >
          <div
            class="c14 c15"
          />
        </div>
      </div>
    </div>
  </section>
  <div
    class="c30"
  >
    <footer
      class="c31"
    >
      <div
        class="c9 c10"
      >
        <div
          class="c14 c15"
        >
          <p>
            teste html
          </p>
        </div>
      </div>
    </footer>
  </div>
  <a
    aria-label="Go to top"
    class="c32"
    href="#"
    title="Go to top"
  >
    <svg
      class="lucide lucide-chevron-up"
      fill="none"
      height="48"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m18 15-6-6-6 6"
      />
    </svg>
  </a>
</div>
`;
